precompiles/staking/approve_test.go

```diff
@@ -195,15 +199,15 @@ func (s *PrecompileTestSuite) TestApprove() {
 			"success - MsgDelegate with 1 Evmos as limit amount",
 			func(_ *vm.Contract) []interface{} {
 				return []interface{}{
-					s.address,
+					s.keyring.GetAddr(0),
 					big.NewInt(1e18),
 					[]string{staking.DelegateMsg},
 				}
 			},
 			func(data []byte, inputArgs []interface{}) {
 				s.Require().Equal(data, cmn.TrueValue)
 
-				authz, expirationTime := s.CheckAuthorization(staking.DelegateAuthz, s.address, s.address)
+				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
```

Comment:
I think the easiest way of improving this (and better readable) would be to define on the package level:

```go
const (
    granterIdx = 0
    granteeIdx = 1
)
```

and then always reference those:

```suggestion
				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(granteeIdx), s.keyring.GetAddr(granterIdx))
```